xdsl/ir.py

```diff
@@ -447,6 +447,20 @@ def __hash__(self) -> int:
         ...
 
 
+@dataclass(frozen=True)
+class OpTrait():
```

Comment:
Could this be in a different file? In terms of import API, I think it would be cool if we had a `traits` file that contains several predefined traits (Pure, NoRegions...)

Reply:
I'm a bit conflicted here.
I would not like to have `OpTrait` in the same file as `Pure`, `NoRegions`, etc... since `OpTrait` is the structure definition, while `Pure` is a definition of it.
In a sense, it would be like putting `Operation` in the same file as `builtin.module`?

What do you think? Should we:
* Put both in `traits.py` (the version you are proposing)
* Put `OpTrait` in `traits.py` and `Pure` in `builtin_traits.py` (a middle ground, but `traits.py` would be quite empty)
* Put `OpTrait` in `ir.py` and `Pure` in `builtin_traits.py` (the current version)